File: Intelligent.OTC.WebApi/common/services/cacheService.js
angular.module('services.cacheService', []);
angular.module('services.cacheService').service('cacheService', ['$cacheFactory', function ($cacheFactory) {

        var service = {};
        var CACHE_KEY = 'OTC';
        var cache = $cacheFactory(CACHE_KEY);

        service.get = function (key) {
            return cache.get(key);
        }

        service.put = function (key, item) {
            cache.put(key, item);
        }

        return service;
    }]);

File: Intelligent.OTC.Business/CacheService.cs
using Intelligent.OTC.Common;
using System;
using System.Web;

namespace Intelligent.OTC.Business
{
    public class CacheService : ICacheService
    {
        public T GetOrSet<T>(string cacheKey, Func<T> getItemCallBack) where T : class
        {
            T item = default(T);
            if (HttpContext.Current != null)
                item = HttpContext.Current.Cache.Get(cacheKey) as T;

            if (item == null)
            {
                item = getItemCallBack();
                if (HttpContext.Current != null)
                    HttpContext.Current.Cache.Insert(cacheKey, item);
            }
            return item;
        }

    }
}
